### PR TITLE
HPCC-22599 Disable Data Patterns for unsupported file types

### PR DESCRIPTION
**Pull request by GordonSmith**
Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [x] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by GordonSmith** [comment]
@miguelvazq please review

**Review by miguelvazq — APPROVED** [review]

File: esp/src/src/DataPatternsWidget.ts [hunk 1]
```diff
@@ -49,6 +49,8 @@ type _TabContainerWidget = {
     addChild(tabItem: any, pos: number);
 };
 
+export const supportedFileType = (contentType: string): boolean => ["flat", "csv", "thor"].indexOf((contentType || "").toLowerCase()) >= 0;
```
Comment (richardkchapman) [inline esp/src/src/DataPatternsWidget.ts]: What about XML?

**Comment by richardkchapman** [comment]
@dcamper Please review

File: esp/src/src/DataPatternsWidget.ts (hunk 1 above)
Comment (dcamper) [reply esp/src/src/DataPatternsWidget.ts]: The script submits a workunit containing generated ECL.  In that ECL is something like:

ds := DATASET(filePath, RECORDOF(filePath, LOOKUP), flat);

In other words, it relies on compile-time record definition lookup.  Is that available/correct with XML files?  My understand is that it was not (always) available and, if not, XML files should be ignored.

Comment (GordonSmith) [reply esp/src/src/DataPatternsWidget.ts]: @dcamper I didn't realize that was the issue - in theory I do have the actual RECORD available to me - if I used that, does it mean XML would also be supported (this is the ECL record as per the Logical File details page).

Comment (dcamper) [reply esp/src/src/DataPatternsWidget.ts]: DataPatterns.Profile() doesn't actually care about the underlying file format at all.  You can even profile a file on the landing zone with a properly-crafted DATASET pathname and record definition.  Bottom line, if you can determine the record layout then by all means, let's use it!  And please give me a hint offline, as I have some other code that needs modifying....

Comment (dcamper) [reply esp/src/src/DataPatternsWidget.ts]: FYI:  I have created a new Jira for the XML case.  As per a discussion I had with Gordon, this PR should be focused on only the undefined filetype case.

**Comment by dcamper** [comment]
@GordonSmith Back to you for XML file type adjustment, whatever that may be.

**Pull request merged**